### DISCUSSION TITLE
Using RS232 I am looking to type a string into a terminal window and for the string to be echoed back to the terminal using micropython

### DISCUSSION THREAD

**Discussion by afrost1402** [micropython/micropython#12208] [Using MicroPython]
Using RS232 I am looking to type a string into a terminal window and for the string to be echoed back to the terminal using micropython. However with the code shown below I get an error with the bytes being unable to convert to a string. Any help would be appreciated. Thanks

`  from machine import UART, Pin, I2C
  import utime
  
  uart = UART(0, baudrate=9600, tx=Pin(0), rx=Pin(1))
  uart.init(bits=8, parity=None, stop=2)
  SDA = Pin(16)
  SCL = Pin(17)
  
  
  uart.write("Script Begin\n")
  
  
  while True:
      data = uart.read()
  
      if data is not None:
          # display input
          print('\nInput:')
          print(data)
          print(type(data))
           
          # converting
          output = data.decode('UTF-8')
           
          # display output
          print('\nOutput:')
          print(output)
          print(type(output))
  
      utime.sleep(3)`

**Comment by karfas** [comment]
And the error message you get is ... ?

The question around converting bytes/bytearray pops up every week or so, e.g. in #11588.
You might be able to find other solutions when searching for "decode" in discussions.

**Reply by afrost1402** [reply]
`Traceback (most recent call last):
  File "<stdin>", line 42, in <module>
UnicodeError: `

Thats the error. I have given that solution a try but I still get the same error message

**Reply by karfas** [reply]
Your data is not valid UTF-8 (and you don't show us the problematic data). 
You will need to use one of the workarounds mentioned in #11588.
